### DISCUSSION TITLE
Require comment functionality on blog

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#12491] [Q&A]
Currently blog theme has no comment functionality 

![image](https://user-images.githubusercontent.com/5202291/192052243-45cb1ffd-e3de-4302-907c-59860f40066f.png)

option over post how I enable that feature or how I implement that, please advise @ns8482e @Skrypt

**Comment by hishamco** [comment]
You can implement it as a content part attached to a blog post

**Reply by saleemhaq** [reply]
> You can implement it as a content part attached to a blog post

Can you please share with the example of how to do that? please

**Reply by hishamco** [reply]
Saleem I suggest to spend some time read in docs or investigate on the code if it's needed. Regarding what you ask you can create a content part named `CommentPart` then add it in the `BlogPost` content type.

If I'm not wrong the same functionality already implemented in Orchard CMS [here](https://github.com/OrchardCMS/Orchard/tree/dev/src/Orchard.Web/Modules/Orchard.Comments)

**Reply by Skrypt** [reply]
Also alternatively you could want to implement a Discuss module for that matter.

**Reply by hishamco** [reply]
If I'm not wrong @agriffard already implemented Discuss module

**Reply by saleemhaq** [reply]
> Saleem I suggest to spend some time read in docs or investigate on the code if it's needed. Regarding what you ask you can create a content part named `CommentPart` then add it in the `BlogPost` content type.
> 
> If I'm not wrong the same functionality already implemented in Orchard CMS [here](https://github.com/OrchardCMS/Orchard/tree/dev/src/Orchard.Web/Modules/Orchard.Comments)

How i put the submit button over there when someone is comment so comment part are saved on orchard and when later any public user come on site so the comment is showing as well, can u please advise me with help of screen shot for clear understanding @hishamco @ns8482e @Skrypt

**Reply by saleemhaq** [reply]
> Also alternatively you could want to implement a Discuss module for that matter.

you mean to say that link module: https://github.com/OrchardCMS/Orchard/tree/dev/src/Orchard.Web/Modules/Orchard.Comments I be used there? @Skrypt  or can you advise me how could i achieve with content part please.

**Reply by hishamco** [reply]
You can add your content part progrmatically or via UI, please spend some time to read the docs or checkout Dojo 3 videos by Lombiq

**Reply by saleemhaq** [reply]
> If I'm not wrong @agriffard already implemented Discuss module

How can I use if it is already implemented, please advise and guide so it help to I success in create blog and their detail just need to put option of comments like below:

![image](https://user-images.githubusercontent.com/5202291/192157518-b581fc0d-b0ae-4351-a976-4c27969c3f8c.png)

so public user able to comment @hishamco @Skrypt @ns8482e @agriffard 
Folks, please help me on above

**Reply by hishamco** [reply]
Try to enable the Discus module into your app

**Reply by hishamco** [reply]
Try to enable the Discus module into your app

**Reply by saleemhaq** [reply]
> Try to enable the Discus module into your app

I try to add that module through the nugget and it successfully done:

![image](https://user-images.githubusercontent.com/5202291/192343158-e32057bf-318d-42f8-b614-32b56fd66313.png)

and do all the configurations that define below see:

- Add the nuget package Disqus.OrchardCore as a reference to your application using Orchard Core.
- Launch your Orchard Core application, login as admin, then go to the Features admin page and enable the Disqus module.
- Add a Disqus Part in the Definition of the Content Type on which you want to add Disqus comments.
- Edit the settings of the Disqus Part and enter the name of the site that you created on https://disqus.com/.
- Do not forget to set a Base Url in the General settings of your site (even if you are testing on localhost).

Add the name part of disquss and do the configuration but still is not show also used the comment module I have the decouple cms

![image](https://user-images.githubusercontent.com/5202291/192516938-1ee4700c-18bb-4ed6-8ea8-37266ac64c40.png)

and the page look like that:

![image](https://user-images.githubusercontent.com/5202291/192517181-020b5815-2c31-4543-8bd8-f3f4caca1452.png)

![image](https://user-images.githubusercontent.com/5202291/192517326-a6d4d2d0-29e7-49e5-adfa-583c9c3cb388.png)

I select i want to comment on side:

![image](https://user-images.githubusercontent.com/5202291/192517371-43ec1ddb-7c7f-4fb2-9f55-93e485ec9742.png)

Please advise @hishamco @sebastienros @Skrypt @ns8482e

**Comment by saleemhaq** [comment]
I just need to know how I handle submit on comment section as later to
manage comment trail? I read the doc's much and go through all videos twice
of time. Just need guidance regarding that.

**Comment by ns8482e** [comment]
@saleemhaq Line 13 doesn’t work as you are you are calling DisplayAsync on json where shape is needed


First build display shape for your content item and then render part shape

```cs
var itemShape = await ContentItemDisplayManager.BuildDisplayAsync(item, updater, "Detail");


@await DisplayAsync(itemShape.Content.PartName)

```

Learn more from here https://docs.orchardcore.net/en/latest/docs/reference/modules/Flow/BagPart/#templating-with-display-management

**Reply by saleemhaq** [reply]
> ```cs
> var itemContent = await ContentItemDisplayManager.BuildDisplayAsync(item, updater, "Detail");
> 
> 
> @await DisplayAsync(itemContent.Content.PartName)
> ```

I try that but not this method come with orchard helper:

![image](https://user-images.githubusercontent.com/5202291/192775543-6bc87fa3-57c1-476e-a89e-310d9952336d.png)

**Reply by ns8482e** [reply]
Learn more from here https://docs.orchardcore.net/en/latest/docs/reference/modules/Flow/BagPart/#templating-with-display-management

**Reply by saleemhaq** [reply]
> @saleemhaq Line 13 doesn’t work as you are you are calling DisplayAsync on json where shape is needed
> 
> First build display shape for your content item and then render part shape
> 
> ```cs
> var itemContent = await ContentItemDisplayManager.BuildDisplayAsync(item, updater, "Detail");
> 
> 
> @await DisplayAsync(itemContent.Content.PartName)
> ```

@ns8482e any luck if you're used that Commentator.OrchardCore in Decouple CMS 
https://github.com/jsalgado10/Commentator.OrchardCore
I tried from my last 4 hours but not luck that you suggest above because its razor page I am try the suggested but not work comments sections not show also try to inject Disqus module it also not work any code need to inject for disquss module to work in razor page?
please advise @Skrypt

**Reply by Skrypt** [reply]
I never used this module so far. I would need to try it. I can't say much. Maybe @agriffard could help.
